src/OptiHPLCHandler/empower_instrument_method.py

```diff
@@ -0,0 +1,99 @@
+import logging
+import re
+
+logger = logging.getLogger(__name__)
+
+
+def instrument_method_factory(method_definition):
+    try:
+        if method_definition["name"] in ["rAcquityFTN"]:
+            logger.debug(f"Creating SampleManager")
+            return SampleManager(method_definition)
+        # Add more cases as they are coded
+        else:
+            logger.debug(
+                f"Unknown instrument method: {method_definition['name']}, "
+                "creating a generic InstrumentMethod"
+            )
+            raise ValueError(f"Unknown instrument method: {method_definition['name']}")
+    except (KeyError, ValueError) as e:
+        if isinstance(e, KeyError):
+            # If the name key is not present, we don't know what to do with it, but we
+            # can still create a generic InstrumentMethod and just return that.
+            logger.debug(f"KeyError: {e}, creating a generic InstrumentMethod")
+        return InstrumentMethod(method_definition)
+
+
+class InstrumentMethod:
+    def __init__(self, method_definition):
+        self.original_method = method_definition
+        self._change_list = []
+
+    def replace(self, original, new):
+        self._change_list.append((original, new))
+
+    @property
+    def current_method(self):
```

Comment:
I would avoid having a getter do too much. A property gives the user the illusion that they are merely reading a variable of an object - as it should be. But for the same reason the user might decide to read it 1 million times by placing it in a loop.

Turn the property into a method?

Reply:
On the other hand, I like that the user doesn't have to know that it is not an existing thing. That is implementation details that the user should not care about.

I would prefer adding caching is performance is an issue. There is a built in way to do that, and we know when to not use the cached version (when self._change_list has changed).

Reply:
As a note to self, if I do need caching, I need to implement __eq__ and __hash__: https://docs.python.org/3/faq/programming.html#faq-cache-method-calls